src/chromo/util.py

```diff
@@ -733,3 +810,94 @@ def select_long_lived(tau=0, mm=False):
             long_lived.append(pid)
 
     return long_lived
+
+
+class redirect_std_streams:
```

Comment:
1) This is a brittle solution, because IPython and Jupyter notebooks also mess around with the streams. Have you tested that this works in IPython and in Jupyter notebooks?

2) Does this solution have confusing side-effects? Can it be asserted that only output from the Fortran code is captured? What if a Python process prints at the same time, e.g. when running multi-threaded?

3) Does this really work? Perhaps Fortran bypasses the Python stdout / stderr?

4) Where are the unit tests for this?

Reply:
Indeed! It's super brittle :((( 

It only works with a model in some standard cases (actually also in Jupyter) but once you start printing from python it looses focus.

I had a Fortran-based mechanism in place to open and close files and maintain output units. That was removed, but in practice it worked quite nicely. So how can I logs, I need them for actual work and also from multiprocessing.

Shall I try reviving the Fortran file management?

Reply:
I think the proper way is to patch the FORTRAN codes that they don't write to stdout but to a file that we specify. I looked into how that can be achieved, but I am not a FORTRAN expert. The Fortran file management, if I remember correctly, was not doing that, it was also a brittle solution similar to this one.

Reply:
What I remember is that WRITE is a statement with a special syntax, so that we cannot just replace write statements with our own function calls. 

I think what we should do is to open our own files and then patch the WRITE statements so that they use our file handle instead of stdout. This patch can probably be applied automatically, so that we don't have to modify the source code in the repository. That would be my preferred solution.

Reply:
If you need this urgently, the redirect_std_streams can be an intermediate solution. But then it should not be public so that we can remove it later (name starts with _).

In any case there need to be some unit tests. The PR title is misleading then. New features are not part of maintenance. And this is not a release PR, or is it?

Reply:
I removed this functionality and will push it to user code. It doesn't really depend on chromo.

Reply:
That's also a solution. To clarify, I think it is fine if you want to place this in chromo's utils, if you use it a lot yourself it might be useful for others as well. But it should be documented as a temporary solution that might disappear in the future.